### DISCUSSION TITLE
How to setup API endpoint at different root than /api

### DISCUSSION THREAD

**Discussion by mikevercoelen** [vercel/next.js#13816] [Help]
Title says it all,

Basically I need to setup webhooks from Stripe, which POST at mydomain.com/webhook

How can I create an API route that's not under /api/webhook but under /webhook

Hope that made sense,

Peace

**Comment by jamesmosier** [comment]
Currently that is not possible with Next's built-in API routes. The API routes always live at `/api/*`

Some options would be to set this redirect up at the server level with your proxy (i.e. nginx or apache or whatever you use). Otherwise [a custom server](https://nextjs.org/docs/advanced-features/custom-server) could help in this scenario but it is strongly discouraged because it will _"remove important performance optimizations, like serverless functions and Automatic Static Optimization."_

**Comment by revskill10** [comment]
Have you tried `rewrites` (it's experimental) ?

**Comment by wereHamster** [comment]
Stripe allows you to use arbitrary webhook URLs. So you can point it to for example https://mydomain.com/api/stripe/webhook.
